src/server/completions/completions.ts

```diff
@@ -46,16 +64,275 @@ export function getCompletionItems(
       },
     };
   });
+  return cleanedCompletions;
 }
 
-export function resolveCompletionItem(item: CompletionItem): CompletionItem {
-  item.detail = item.data.property;
-  const docs = (COMPLETION_DOCS[item.data.type] || {})[item.data.property];
-  if (docs) {
-    item.documentation = {
-      kind: MarkupKind.Markdown,
-      value: docs,
+async function getSchemaCompletions(
+  document: TextDocument,
+  context: CompletionParams,
+  translateCache: TranslateCache
+): Promise<string[]> {
+  const parse = parseDocumentText(document.getText(), context.position);
+  if (parse.nearestDefinitionText && parse.adjustedCursor) {
+    const exploreName = getQueriedExploreName(parse.nearestDefinitionText);
+    const {keyword, chain} = getQueryContext(
+      parse.nearestDefinitionText,
+      parse.adjustedCursor
+    );
+    if (exploreName !== null && keyword != null && chain !== null) {
+      let model: Model | undefined = undefined;
+      try {
+        model = await translateCache.translateWithTruncatedCache(
+          document,
+          parse.truncatedText,
+          parse.exploreCount,
+          exploreName
+        );
+      } catch (error: unknown) {
+        // eslint-disable-next-line no-console
+        console.error(
+          `Error fetching model for document sources and imports '${__filename}': ${error}`
+        );
+      }
+      if (model) {
+        const exploreMap: Record<string, Explore> = {};
+        model.explores.forEach(explore => {
+          exploreMap[explore.name] = explore;
+        });
+        if (exploreMap[exploreName]) {
+          const fields = getFieldsFromChain(
+            chain,
+            exploreMap[exploreName],
+            exploreMap
+          );
+          return filterCompletions(fields, keyword, chain);
+        }
+      }
+    }
+  }
+  return [];
+}
+
+function getQueriedExploreName(text: string[]): string | null {
+  const unnamedQuery =
+    /\b(run|query)\s*:\s*([A-Za-z_][A-Za-z_0-9]*)\s*->\s*{/gi;
+  const unnamedQueryMatch = unnamedQuery.exec(text[0]);
+  if (unnamedQueryMatch) {
+    return unnamedQueryMatch[2];
+  }
+  const namedQuery =
+    /\bquery\s*:\s*([A-Za-z_][A-Za-z_0-9]*)\s+is\s+([A-Za-z_][A-Za-z_0-9]*)\s+->\s*{/gi;
+  const namedQueryMatch = namedQuery.exec(text[0]);
+  return namedQueryMatch ? namedQueryMatch[2] : null;
+}
+
+function getQueryContext(lines: string[], cursor: Position) {
+  lines[cursor.line] = lines[cursor.line].slice(0, cursor.character);
+  const chainRegex =
+    /\s+(([a-zA-z_][A-Za-z_0-9]*)(\.([a-zA-z_][A-Za-z_0-9]*)*)*)?$/g;
```

Comment:
Not sure about the backtracking, but the other complaints are from the `A-z` (lowercase Z)

Reply:
Working on the inefficient regex warning, resolved the A-z issue